rust/src/main.rs

```diff
@@ -0,0 +1,20 @@
+use nitor_vault::Vault;
+
+#[tokio::main]
+async fn main() {
+    println!("Nitor Vault, see https://github.com/nitorcreations/vault for usage examples");
+    println!("testing listing all vault items...");
+    let client = Vault::new(None, None).await.expect("Error getting Vault");
+    list_all(&client).await;
+    // client
+    //     .store("testingtesting", "hello team".as_bytes())
+    //     .await;
+    // println!("{}", client.lookup("testingtesting").await.unwrap())
```

Comment:
You have some commented out code here, could clean this up 😉

Reply:
Yeah, the main.rs shouldn't even be there yet as it should be a CLI, but didn't remove it. the whole file probably should be removed.

Reply:
Although the current main function could have its place in the Readme for example usage. 🤔

Reply:
I added the [clap](https://docs.rs/clap/latest/clap/) library now so could build the CLI on top of that. Internet says it is the most popular CLI lib for rust. I can start working on that if you would like 🤓

Reply:
Go for it! 👍

Reply:
Basic CLI up and running:
```console
➜  rust git:(rust) ./target/release/vault -V
nitor-vault 0.1.0
➜  rust git:(rust) ./target/release/vault -h
Encrypted AWS key-value storage

Usage: vault [OPTIONS]

Options:
  -l, --lookup <LOOKUP>  Print secret value for given name
  -a, --all              List available secrets
  -h, --help             Print help (see more with '--help')
  -V, --version          Print version
➜  rust git:(rust) ./target/release/vault -l testingtesting
Loading value for: testingtesting
hello team
➜  rust git:(rust) ./target/release/vault -a
DigiCertGlobalRootCA.crt.pem
__star__.nitor.zone.chain
__star__.nitor.zone.chained.crt
__star__.nitor.zone.crt
__star__.nitor.zone.key.clear
accounting-database.json
activegate.nitor.zone-ssh-hostkeys.sh
amibakery.nitor.zone.chain
amibakery.nitor.zone.chained.crt
amibakery.nitor.zone.crt
```

Reply:
Nice! would probably need -s also and then the CLI could also be included in the release. But good job 👌

Reply:
Sure 👍🏻 Sorry, I am used to all my work being in private repos and I did not realize the key names can be sensitive info...